gammapy/datasets/utils.py

```diff
@@ -30,7 +31,7 @@ def apply_edisp(input_map, edisp):
         energy_axis = input_map.geom.axes["energy_true"].copy(name="energy")
 
     geom = input_map.geom.to_image().to_cube(axes=[energy_axis])
-    return input_map._init_copy(geom=geom, data=data)
+    return Map.from_geom(geom=geom, data=data, unit=input_map.unit)
```

Comment:
I think `self.__class__(geom, data, unit, ...)` should even be faster...

Reply:
done